### PR TITLE
Cleaned up `CreatePlrItems`.

### PR DESCRIPTION
**Pull request by seritools**
Not much to say here, pretty straight-forward and easily testable.

TODOs and remaining differences added as comments.

### CODE REVIEW

**Comment by seritools** [comment]
Now 99% accurate (can't get the last `GetPlrHandSeed` call in the switch to optimize the way it is in the binary, TODO for that added) ;)

File: Source/items.cpp [hunk 1]
```diff
@@ -1519,83 +1519,94 @@ void __fastcall SetPlrHandGoldCurs(ItemStruct *h)
...
-	qmemcpy((char *)&plr[0].InvList[plr[v1]._pNumInv++] + v1 * 21720, &plr[v1].HoldItem, 0x170u);
-	plr[v1].InvGrid[30] = plr[v1]._pNumInv;
-	CalcPlrItemVals(player_numa, 0);
+
+	SetPlrHandItem(&plr[p].HoldItem, IDI_GOLD);
+	GetPlrHandSeed(&plr[p].HoldItem);
+
+	// TODO: Add debug logic from 1.00 here
+
+	plr[p].HoldItem._iCurs = 4;
```
Comment (mewmew) [inline Source/items.cpp]: Is there an enum for this 4?

File: Source/items.cpp [hunk 2]
```diff
@@ -1519,83 +1519,94 @@ void __fastcall SetPlrHandGoldCurs(ItemStruct *h)
...
-			SetPlrHandItem(&plr[v1].SpdList[1], IDI_MANA);
-LABEL_14:
-			GetPlrHandSeed(&plr[v1].SpdList[1]);
+
+	// TODO: The compiler seems to load the argument for the last call
+	// of `GetPlrHandSeed` twice instead of once
+	switch ( plr[p]._pClass )
+	{
+		case PC_SORCERER:
+			SetPlrHandItem(&plr[p].InvBody[4], IDI_SORCEROR);
```
Comment (mewmew) [inline Source/items.cpp]: I'm wondering if there may be an enum for InvBody indices. E.g. 4 may be right hand, or so.

File: Source/items.cpp (hunk 1 above)
Comment (seritools) [reply Source/items.cpp]: Yeah, you're right. It's probably `cursor_id::CURSOR_RECHARGE` :)

File: Source/items.cpp (hunk 2 above)
Comment (seritools) [reply Source/items.cpp]: Probably, I don't think there is one defined yet. Do you or @galaxyhaxz have the correct mapping for that?

Comment (mewmew) [reply Source/items.cpp]: > Probably, I don't think there is one defined yet. Do you or @galaxyhaxz have the correct mapping for that?

I had a look but didn't find any current definition for it.

**Comment by seritools** [comment]
Now binary exact with the new compiler version.

**Pull request merged**